service/views.py

```diff
@@ -1,26 +1,69 @@
 import json
 
-from django.http import JsonResponse
+from django.http import HttpResponse, JsonResponse
 from decouple import config
 from lib_version_remla24_team02 import VersionUtil
+from prometheus_client import Counter, Histogram, Gauge, generate_latest
+
 import requests
 
+request_count = Counter('request_count', 'Total number of requests')
+request_duration = Histogram('request_duration_milliseconds', 'Duration of request processing in seconds', buckets=[0.00001, 0.00005, 0.0001, 0.0005, 0.001, 0.01, 0.1, 1])
+phishing_url_count = Counter('phishing_url_count', 'Total number of phishing URLs detected')
+safe_url_count = Counter('safe_url_count', 'Total number of safe URLs detected')
+prediction_accuracy = Gauge('prediction_accuracy', 'Accuracy of the predictions based on user feedback')
+user_feedback_correct = Counter('user_feedback_correct', 'Total number of correct user feedback')
+user_feedback_incorrect = Counter('user_feedback_incorrect', 'Total number of incorrect user feedback')
 
 def version(request):
     return JsonResponse({'version': f"v{VersionUtil.get_version()}"})
 
 
 def detect(request):
     if request.method == 'POST':
-        # Get the url from the frontend request
+        request_count.inc() # Increment request counter
+
+        with request_duration.time(): # Time the request
+            # Get the url from the frontend request
+            body = json.loads(request.body.decode('utf-8'))
+            url = body.get('url')
+
+            # Forward the request to the model-service and get the response (as int) from them
+            response = requests.post(f"{config('MODEL_SERVICE_URL')}/api/predict",
+                                    headers={'token': config('API_KEY')}, json={'url': url})
+            prediction = response.json().get('prediction')
+
+            # Convert the prediction into a safe vs phishing detection
+            safe = prediction == 0
+
+            if safe:
+                safe_url_count.inc()
+            else:
+                phishing_url_count.inc()
+            
+            return JsonResponse({'safe': safe})
+        
+
+def feedback(request):
+    if request.method == 'POST':
         body = json.loads(request.body.decode('utf-8'))
         url = body.get('url')
```

Comment:
You aren't using this url in the parameters you store or anywhere else in this method. In case you change it then remember to also remove this from the frontend as not doing so won't give an error but will send data unnecessarily which is a waste.

A possible use for this might be to only report the metrics using unique urls, if the same url is requested multiple times, or is given feedback multiple times then nothing will be stored on later runs. Got it's ups and downs, just might be something